lib/luacrypto_comp.c

```diff
@@ -0,0 +1,135 @@
+/*
+* SPDX-FileCopyrightText: (c) 2025 jperon <cataclop@hotmail.com>
+* SPDX-License-Identifier: MIT OR GPL-2.0-only
+*/
+
+/***
+* Low-level Lua interface to the Linux Kernel Crypto API for synchronous
+* compression algorithms.
+*
+* This module provides a `new` function to create COMP transform objects,
+* which can then be used for compression and decompression.
+*
+* @module crypto_comp
+*/
+
+#define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
+
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/crypto.h>
+#include <linux/err.h>
+#include <linux/slab.h>
+#include <linux/limits.h>
+
+#include <lua.h>
+#include <lualib.h>
+#include <lauxlib.h>
+#include <lunatik.h>
+
+#include "luacrypto.h"
+
+LUNATIK_PRIVATECHECKER(luacrypto_comp_check, struct crypto_comp *);
+
+LUACRYPTO_RELEASER(comp, struct crypto_comp, crypto_free_comp, NULL);
+
+/***
+* COMP Object methods.
+* These methods are available on COMP objects created by `crypto_comp.new()`.
+* @type crypto_comp
+*/
+
+#define LUACRYPTO_COMP_OPERATION(name)									\
+static int luacrypto_comp_##name(lua_State *L) {							\
+	struct crypto_comp *tfm = luacrypto_comp_check(L, 1);						\
+	size_t datalen;											\
+	const u8 *data = (const u8 *)luaL_checklstring(L, 2, &datalen);					\
+	lunatik_checkbounds(L, 2, datalen, 0, UINT_MAX);							\
```

Comment:
```suggestion
	lunatik_checkbounds(L, 2, datalen, 1, UINT_MAX);						\
```

please, mind the `\` alignment